fv3gfs-util/tests/test_partitioner.py

```diff
@@ -108,150 +107,436 @@ def test_tile_extent_from_rank_metadata(array_extent, array_dims, layout, tile_e
+            (5, 4, 4),
+            (2, 3),
+            3,
+            (slice(0, 5), slice(2, 4), slice(0, 1)),
+            False,
+            0.5,
+            id="36_rank_topleft_no_overlap_rectangle_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 4, 4),
+            (2, 3),
+            4,
+            (slice(0, 5), slice(2, 4), slice(1, 3)),
+            False,
+            0.5,
+            id="36_rank_topmid_no_overlap_rectangle_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 4, 4),
+            (2, 3),
+            5,
+            (slice(0, 5), slice(2, 4), slice(3, 4)),
+            False,
+            0.5,
+            id="36_rank_topright_right_no_overlap_rectangle_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            0,
+            (slice(0, 5), slice(0, 2), slice(0, 2)),
+            False,
+            0.5,
+            id="54_rank_botleft_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            1,
+            (slice(0, 5), slice(0, 2), slice(2, 6)),
+            False,
+            0.5,
+            id="54_rank_botmid_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            4,
+            (slice(0, 5), slice(2, 6), slice(2, 6)),
+            False,
+            0.5,
+            id="54_rank_midmid_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            5,
+            (slice(0, 5), slice(2, 6), slice(6, 8)),
+            False,
+            0.5,
+            id="54_rank_midright_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            8,
+            (slice(0, 5), slice(6, 8), slice(6, 8)),
+            False,
+            0.5,
+            id="54_rank_topright_right_no_overlap_square_layout_half_edge_tiles_3d",
+        ),
+        pytest.param(
+            [fv3gfs.util.Z_INTERFACE_DIM, fv3gfs.util.Y_DIM, fv3gfs.util.X_DIM],
+            (5, 8, 8),
+            (3, 3),
+            0,
+            (slice(0, 5), slice(0, 1), slice(0, 1)),
+            False,
+            float(1 / 6),
```

Comment:
FYI: Integer division always returns a float in python 3, but you can be really sure by using a dot to make one of the numbers a float like in `1./6`.

Reply:
Done